contracts/src/spaces/facets/membership/MembershipBase.sol

```diff
@@ -109,7 +109,20 @@ abstract contract MembershipBase is IMembershipBase {
   }
 
   function _getCreatorBalance() internal view returns (uint256) {
-    return MembershipStorage.layout().tokenBalance;
+    MembershipStorage.Layout storage ds = MembershipStorage.layout();
+    uint256 contractBalance = address(this).balance;
+
+    // Rebalance stored balance if it doesn't match actual balance
+    // only if the contract balance is less than the stored balance
```

Comment:
What if the contract has surplus ETH? Does it have a `fallback` function?

Reply:
The only way it would have surplus ETH apart from just sending directly is if there's an entitlement check that's pending

Reply:
no fallback for now